runners/flink/src/main/java/org/apache/beam/runners/flink/LyftFlinkStreamingPortableTranslations.java

```diff
@@ -193,33 +225,63 @@ public boolean isEndOfStream(WindowedValue<byte[]> nextElement) {
   }
 
   // TODO: translation assumes byte[] values, does not support keys and headers
-  private void translateKafkaSink(
+  @VisibleForTesting
+  void translateKafkaSink(
       String id, RunnerApi.Pipeline pipeline, StreamingTranslationContext context) {
     RunnerApi.PTransform pTransform = pipeline.getComponents().getTransformsOrThrow(id);
     final String topic;
-    final Properties properties = new Properties();
+    final String bootstrapServers;
+
     ObjectMapper mapper = new ObjectMapper();
+    FlinkKafkaProducer011<WindowedValue<byte[]>> producer;
+    LyftKafkaProducerBuilder<WindowedValue<byte[]>> producerBuilder =
+        new LyftKafkaProducerBuilder<>();
     try {
       Map<String, Object> params =
           mapper.readValue(pTransform.getSpec().getPayload().toByteArray(), Map.class);
 
       Preconditions.checkNotNull(topic = (String) params.get("topic"), "'topic' needs to be set");
 
-      Map<?, ?> consumerProps = (Map) params.get("properties");
-      Preconditions.checkNotNull(consumerProps, "'properties' need to be set");
-      properties.putAll(consumerProps);
+      Map<?, ?> producerProps = (Map) params.get("properties");
+      Preconditions.checkNotNull(producerProps, "'properties' need to be set");
+      logger.info(
+          "Configuring kafka producer for topic {} with properties {}", topic, producerProps);
+
+      Preconditions.checkNotNull(
+          bootstrapServers = (String) producerProps.remove("bootstrap.servers"),
+          "'bootstrap.servers' needs to be set");
+
+      final String splits[] = bootstrapServers.split(":", 2);
+      final String bootstrapServer = splits[0];
+      int port = splits.length == 2 ? Integer.parseInt(splits[1]) : DEFAULT_KAFKA_BROKER_PORT;
+      final String userName = (String) params.get("username");
+      final String password = (String) params.get("password");
+
+      if (userName != null) {
+        producerBuilder.withUsername(userName);
+      }
+
+      if (password != null) {
+        producerBuilder.withPassword(password);
+      }
+
+      for (Map.Entry<?, ?> entry : producerProps.entrySet()) {
+        producerBuilder.withKafkaProperty(
+            String.valueOf(entry.getKey()), String.valueOf(entry.getValue()));
+      }
+
+      producer =
```

Comment:
Since the user has already added `bootstrap.servers` to the properties and we pass all the properties to the builder, why do we need to extract it here rather than doing that inside the builder?

Reply:
Given,  the required params are passed explicitly to the `build` method 

https://github.com/lyft/streamingplatform/blob/STRMCMP-835-a/spflinkjava/streamingplatform-kafka/src/main/java/com/lyft/streamingplatform/LyftKafkaProducerBuilder.java#L96.

We can go with overloaded `build` methods or just simplify the build method as below

1. ProducerBuilder  `build(topic, serializationSchema)`  
2. ConsumerBuilder `build(topic, kafkaDeserializationSchema)`

Reply:
I think an overloaded build method that also takes the properties would be fine to keep things simple for the case where the properties, as expected by the Kafka client, are already provided. The other flavor may be useful when that is not the case.

Reply:
Given we have added `withKafkaProperties` https://github.com/lyft/streamingplatform/blob/7549a1bec51407c74a1af529cecad1ff8ddaa75c/spflinkjava/streamingplatform-kafka/src/main/java/com/lyft/streamingplatform/LyftKafkaConsumerBuilder.java#L42 ,  
we should avoid having an overloaded build method that takes Properties. 

All kafka client properties will be passed either through `withKafkaProperty` or `withKafkaProperties` .   WDYT ?  If you downvote, I am happy to make the change you proposed above.

Reply:
Sounds good.